extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts

```diff
@@ -113,11 +113,17 @@ function generateMarkdownForCodeSnippet(
     .map((line, index) =>
       highlightCodeLines(line, index + snippetStartLine, highlightedRegion)
     );
-  lines.push(
-    `<pre><code class="${language}">`,
-    ...codeLines,
-    '</code></pre>',
-  );
+
+  // Make sure there are no extra newlines before or after the <code> block:
+  if (codeLines.length === 1) {
+    lines.push(`<pre><code class="${language}">${codeLines[0]}</code></pre>`);
+  } else {
+    lines.push(
+      `<pre><code class="${language}">${codeLines[0]}`,
+      ...codeLines.slice(1, -1),
+      `${codeLines[codeLines.length - 1]}</code></pre>`,
+    );
+  }
```

Comment:
This might work. It's maybe 10% better. 

```suggestion
  // Make sure there are no extra newlines before or after the <code> block:
  const codeLinesWrapped = [...codeLines];
  codeLinesWrapped[0] = `<pre><code class="${language}">${codeLinesWrapped[0]}`;
  codeLinesWrapped[codeLinesWrapped.length - 1] = `${codeLinesWrapped[codeLinesWrapped.length - 1]}</code></pre>`;
  
  lines.push(...codeLinesWrapped);
```